api/server/server.go

```diff
@@ -177,26 +180,35 @@ func (s *Server) Start(ctx context.Context) {
 }
 
 func (s *Server) startGears(ctx context.Context) {
+	// By default it serves on :8080 unless a
+	// PORT environment variable was defined.
+	listen := fmt.Sprintf(":%d", viper.GetInt(EnvPort))
+	listener, err := net.Listen("tcp", listen)
+	if err != nil {
+		logrus.WithError(err).Fatalln("Failed to serve functions API.")
+	}
+	logrus.Infof("Serving Functions API on address `%s`", listen)
+
 	svr := &supervisor.Supervisor{
 		MaxRestarts: supervisor.AlwaysRestart,
 		Log: func(msg interface{}) {
 			logrus.Debug("supervisor: ", msg)
 		},
 	}
 
-	// By default it serves on :8080 unless a
-	// PORT environment variable was defined.
 	svr.AddFunc(func(ctx context.Context) {
-		go s.Router.Run()
+		http.Serve(listener, s.Router)
```

Comment:
How is this terminated after context cancelation? `http.Serve` is blocking and it is never going to be stopped, unless it panics or fails binding the port. And in that case, also you are not catching the error...